service/src/main/java/bio/terra/workspace/service/resource/controlled/ControlledResourceService.java

```diff
@@ -189,6 +191,74 @@ public ControlledBigQueryDatasetResource createBqDataset(
     return jobBuilder.submitAndWait(ControlledBigQueryDatasetResource.class);
   }
 
+  /**
+   * Make a clone of a BigQuery dataset
+   *
+   * @param sourceWorkspaceId - workspace ID of original dataset
+   * @param sourceResourceId - resource ID of original dataset
+   * @param destinationWorkspaceId - destination (sink) workspace ID
+   * @param jobControl - job control structure (should already have ID)
+   * @param userRequest - request object for this call
+   * @param destinationResourceName - resource name. Uses source name if null
+   * @param destinationDescription - description string for cloned dataset. Source description if
+   *     null.
+   * @param destinationDatasetName - name for new resource. Can equal source name. If null, a random
+   *     name will be generated
+   * @param destinationLocation - location override. Uses source location if null
+   * @param cloningInstructionsOverride - Cloning instructions for this clone operation, overriding
+   *     any existing instructions. Existing instructions are used if null.
+   * @return
+   */
+  public String cloneGcsBigQueryDataset(
```

Comment:
Probably not `Gcs`.  If you want it to be Gcp, then maybe you also want `createGcpBigQueryDataset` on line 178?